azure-devops/azext_devops/dev/repos/pull_request.py

```diff
@@ -436,6 +437,29 @@ def add_pull_request_work_items(id, work_items, organization=None, detect=None):
     return wit_client.get_work_items(ids=ids)
 
 
+def checkout(id):  # pylint: disable=redefined-builtin
+    """Checkout the PR source branch locally, if no local changes are present
+    :param id: ID of the pull request.
+    :type id: int
+    """
+    organization = resolve_instance(detect='on', organization=None)
+    if not organization:
+        raise CLIError('this command should be used from a valid git repository only')
+
+    client = get_git_client(organization)
+    pr = client.get_pull_request_by_id(id)
+
+    # favorite the ref
+    refFavoriteRequest = GitRefFavorite(name=pr.source_ref_name, repository_id=pr.repository.id, type=2)
+    try:
+        client.create_favorite(favorite=refFavoriteRequest, project=pr.repository.project.id)
+    except Exception as ex:  # pylint: disable=broad-except
+        if 'is already a favorite for user' not in str(ex):
+            raise Exception(ex)
```

Comment:
Should we handle the else part as well or this will end up throwing a stack trace.

Reply:
nope.
it will be handled by the generic exception handler [azure_devops_exception_handler](https://github.com/Microsoft/azure-devops-cli-extension/blob/master/azure-devops/azext_devops/dev/common/exception_handler.py)